### PR TITLE
Analysis Fixture

### LINKED ISSUE #85: Analysis fixture
Add a fixture `analysis` that has the following API:

#### `id`
The analysis database ID

#### `set_result()`
A method that sets the result field of the analysis.
- can only be called once
- takes a dict
- also sets `ready` to `true` on the analysis document
- should take effect after all files are uploaded (analysis is truly ready for viewing)

#### `upload_file()`
A method that flags a working file for upload to the server at the end of the workflow run.
- arguments
  - `name`: the name displayed in the UI to the user
  - `description`: optional description for the file
  - `path`: the path to the file to upload
  - `format`: the 
- other features
  - determine and report the size of the file

Please comment @BlakeASmith

### PR DESCRIPTION
**Pull request by BlakeASmith**
Adds a new fixture `analysis` and abstract base class `AbstractFileUploader`.

The `upload_file` method of `analysis` uses a `DirectoryFileUploder` which only moves the marked 
files to `data_path/uploads`.

Resolves #85

Depends on #90

### CODE REVIEW

**Comment by igboyes** [comment]
Please resolve merge conflicts.

**Comment by BlakeASmith** [comment]
> Please resolve merge conflicts.

Resolved.

File: virtool_workflow/uploads/files.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Literal
+
+
+VirtoolFileFormat = Literal[
```
Comment (igboyes) [inline virtool_workflow/uploads/files.py:6]: Please change to:
- `sam`
- `bam`
- `fasta`
- `fastq`
- `csv`
- `tsv`
- `json`

We should also have an "unknown" format.

File: virtool_workflow/db/db.py [hunk 2]
```diff
@@ -75,3 +74,36 @@ async def store_result(self, id_: str, result: Dict[str, Any], collection: str,
         """Store the result onto the document specified by `id_` in the collection specified by `collection`."""
         await self.db.store_result(id_, self.db[collection], result, file_results_location)
 
+    async def _generate_file_id(self, filename: str) -> str:
```
Comment (igboyes) [inline virtool_workflow/db/db.py:77]: Drop this. Only used for user uploads.

**Review by igboyes — CHANGES_REQUESTED** [review]
I am worried a bit about the case where a workflow developer tries to upload two files with the same name. We may want an `id` field as well with an integer prepended to the name. This would also be the file's name in the application data location.

File: virtool_workflow/abc/db.py [hunk 3]
```diff
@@ -40,3 +41,9 @@ async def store_result(self, id_: str, result: Dict[str, Any], collection: str,
         """Store the result onto the document specified by `id_` in the collection specified by `collection`."""
         ...
 
+    async def create_upload_document(self, file_upload: FileUpload, reserved: bool = False):
+        """Create a document in the database for a file upload."""
+        ...
```
[the remark this answers is not in the document]
Comment (BlakeASmith) [reply virtool_workflow/abc/db.py:46]: No it can be removed.

**Review by igboyes — CHANGES_REQUESTED** [review]
One more change...

**Review by igboyes — APPROVED** [review]
Thanks for persevering!

**Pull request merged**